### PR TITLE
[addons/networking.cilium.io] enable prometheus scraping

### PR DESCRIPTION
**Pull request by ulfox**
When EnablePromethesMetrics is enabled the
required metrics endpoints are exposed but
no annotations is included in the agents.

This commit includes the prometheus.io/scrape
and prometheus.io/port annotations that
prometheus scrappers can use for auto-discovery

Co-authored-by: tchatzig <t.chatzigeorgiou@thebeat.co>

### CODE REVIEW

**Comment by ulfox** [comment]
signed

File: upup/models/cloudup/resources/addons/networking.cilium.io/k8s-1.12-v1.8.yaml.template [hunk 1]
```diff
@@ -417,6 +417,12 @@ spec:
         # gets priority scheduling.
         # https://kubernetes.io/docs/tasks/administer-cluster/guaranteed-scheduling-critical-addon-pods/
         scheduler.alpha.kubernetes.io/critical-pod: ""
+        # Annotation required for prometheus auto-discovery scrapping
```
Comment (johngmyers) [inline upup/models/cloudup/resources/addons/networking.cilium.io/k8s-1.12-v1.8.yaml.template:420]:
```suggestion
        # Annotation required for prometheus auto-discovery scraping
```

File: upup/models/cloudup/resources/addons/networking.cilium.io/k8s-1.12-v1.9.yaml.template [hunk 2]
```diff
@@ -503,6 +503,12 @@ spec:
         # gets priority scheduling.
         # https://kubernetes.io/docs/tasks/administer-cluster/guaranteed-scheduling-critical-addon-pods/
         scheduler.alpha.kubernetes.io/critical-pod: ""
+        # Annotation required for prometheus auto-discovery scrapping
```
Comment (johngmyers) [inline upup/models/cloudup/resources/addons/networking.cilium.io/k8s-1.12-v1.9.yaml.template:506]:
```suggestion
        # Annotation required for prometheus auto-discovery scraping
```

File: upup/models/cloudup/resources/addons/networking.cilium.io/k8s-1.16-v1.10.yaml.template [hunk 3]
```diff
@@ -519,6 +519,12 @@ spec:
         # gets priority scheduling.
         # https://kubernetes.io/docs/tasks/administer-cluster/guaranteed-scheduling-critical-addon-pods/
         scheduler.alpha.kubernetes.io/critical-pod: ""
+        # Annotation required for prometheus auto-discovery scrapping
```
Comment (johngmyers) [inline upup/models/cloudup/resources/addons/networking.cilium.io/k8s-1.16-v1.10.yaml.template:522]:
```suggestion
        # Annotation required for prometheus auto-discovery scraping
```

**Review by johngmyers — CHANGES_REQUESTED** [review]

File: upup/models/cloudup/resources/addons/networking.cilium.io/k8s-1.12-v1.8.yaml.template [hunk 4]
```diff
@@ -417,6 +417,12 @@ spec:
         # gets priority scheduling.
         # https://kubernetes.io/docs/tasks/administer-cluster/guaranteed-scheduling-critical-addon-pods/
         scheduler.alpha.kubernetes.io/critical-pod: ""
+        # Annotation required for prometheus auto-discovery scraping
+        # https://docs.cilium.io/en/v1.9/operations/metrics/#installation
+        {{ if .EnablePrometheusMetrics }}
```
Comment (johngmyers) [inline upup/models/cloudup/resources/addons/networking.cilium.io/k8s-1.12-v1.8.yaml.template:422]: This is outside of any `{{ with .Networking.Cilium }}` block.

**Review by johngmyers — APPROVED** [review]
Thanks!

**Pull request merged**